packages/explorer/src/app/(explorer)/api/system-abis/route.ts

```diff
@@ -0,0 +1,42 @@
+import { Abi, Hex } from "viem";
+import { getSystemAbis } from "@latticexyz/store-sync/internal";
+import { validateChainId } from "../../../../common";
+import { getClient } from "../utils/getClient";
+import { getIndexerUrl } from "../utils/getIndexerUrl";
+
+export const dynamic = "force-dynamic";
+
+export type SystemAbisResponse = {
+  abis: {
+    [systemId: Hex]: Abi;
+  };
+};
+
+export async function GET(req: Request) {
+  const { searchParams } = new URL(req.url);
+  const chainId = Number(searchParams.get("chainId"));
+  const worldAddress = searchParams.get("worldAddress") as Hex;
+  const systemIds = searchParams.get("systemIds")?.split(",") as Hex[];
+
+  if (!chainId || !worldAddress || !systemIds) {
+    return Response.json({ error: "Missing chainId, worldAddress or systemIds" }, { status: 400 });
+  }
+  validateChainId(chainId);
+
+  try {
+    const client = await getClient(chainId);
+    const indexerUrl = getIndexerUrl(chainId);
+    const abis = await getSystemAbis({
+      client,
+      worldAddress,
+      systemIds,
```

Comment:
instead of asking the indexer for system IDs, then passing them in here, then fetching all system resource tags, then filtering based on passed system IDs
why don't we make this param optional and return all system ABIs that are found in the resource tags table?
that way we don't have to query this table twice in two different ways

Reply:
alternatively, since these are internal APIs, we could return all ABIs in one response instead of two juggling separate API calls and hooks?